File: N/2125-number-of-laser-beams-in-a-bank/main.go
package main

func numberOfBeams(bank []string) int {
	prevIdx := -1
	prev := 0
	ret := 0
	for i, l := range bank {
		cnt := 0
		for _, b := range l {
			if byte(b) == byte('1') {
				cnt++
			}
		}
		if cnt > 0 {
			if i-prevIdx > 0 {
				ret += prev * cnt
			}
			prevIdx = i
			prev = cnt
		}
	}
	return ret
}

File: N/2125-number-of-laser-beams-in-a-bank/main.rs
impl Solution {
  pub fn number_of_beams(bank: Vec<String>) -> i32 {
    bank
      .iter()
      .fold((0, 0), |(ans, prev), b| {
        let c = b.chars().filter(|&x| x == '1').count() as i32;
        (ans + prev * c, if c == 0 { prev } else { c })
      })
      .0
  }
}
